packages/oidc/src/get-vercel-oidc-token.ts

```diff
@@ -22,7 +26,24 @@ import { getContext } from './get-context';
  * ```
  */
 export async function getVercelOidcToken(): Promise<string> {
-  return getVercelOidcTokenSync();
+  let token = '';
+  let err: any;
+
+  try {
+    token = getVercelOidcTokenSync();
+  } catch (error) {
+    err = error;
+  }
+
+  try {
+    if (!token || isExpired(getTokenPayload(token))) {
+      await refreshToken();
+      token = getVercelOidcTokenSync();
+    }
+  } catch (error) {
+    throw new Error(`Failed to refresh OIDC token: ${err}: ${error}`);
```

Comment:
One option here (sorry, I cannot remember if I suggested it before and we resolved it):

```suggestion
    throw new Error('Failed to refresh OIDC token', { cause: error });
```

Reply:
oh whoops, you did suggest it and I resolved it after introducing the custom error type but didn't track that it was the other file. I'm good with the suggestion!

Reply:
Ahhh ok I thought I had tried it before making the custom error. There's no options bag on the error constructor pre ES2022 and this module uses ES2021. Is this something we would normally just bump?

Reply:
Hmm, not sure. Maybe we should bump in a different PR. Until then, maybe our errors should use the format you already have 👍